token-metadata/program/src/instruction.rs

```diff
@@ -0,0 +1,120 @@
+use {
+    borsh::{BorshDeserialize, BorshSerialize},
+    solana_program::{
+        instruction::{AccountMeta, Instruction},
+        pubkey::Pubkey,
+        sysvar,
+    },
+    std::str::FromStr,
+};
+
+#[repr(C)]
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug, Clone)]
+/// Args for Create call
+pub struct CreateMetadataAccountArgs {
+    /// The name of the asset
+    pub name: String,
+    /// The symbol for the asset, ie, AAPL or SHOES
+    pub symbol: String,
+    /// URI pointing to JSON representing the asset
+    pub uri: String,
+}
+
+#[repr(C)]
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug, Clone)]
+/// Args for init call
+pub struct InitMetadataAccountArgs {
+    /// The name of the asset
+    pub name: String,
+    /// The symbol for the asset, ie, AAPL or SHOES
+    pub symbol: String,
+    /// URI pointing to JSON representing the asset
+    pub uri: String,
+}
+
+#[repr(C)]
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug, Clone)]
+/// Args for update call
+pub struct UpdateMetadataAccountArgs {
+    /// URI pointing to JSON representing the asset
+    pub uri: String,
+}
+
+/// Instructions supported by the Metadata program.
+#[derive(BorshSerialize, BorshDeserialize, Clone)]
+pub enum MetadataInstruction {
+    /// Create an  Owner and  Metadata objects.
+    ///   0. `[writable]`  Owner key (pda of ['metadata', program id, name, symbol])
+    ///   1. `[writable]`  metadata key (pda of ['metadata', program id, mint id])
+    ///   2. `[]` Mint of token asset
+    ///   3. `[signer]` Mint authority
+    ///   4. `[signer]` payer
+    ///   5. `[]` System program
+    CreateMetadataAccounts(CreateMetadataAccountArgs),
```

Comment:
Why plural?  `CreateMetadataAccounts`

But I think we can just call these two instructions `Create` and `Update`.  "Metadata" is already in the enum name

Reply:
I think because we are creating two different structs, it should be plural.